proposals/xxxx-metadata-providers.md

```diff
@@ -66,35 +113,130 @@ let logger = Logger(label: "example", metadataProvider: { baggage in
 
 ### Task-local Baggage
 
-Once configured, the `log` function and its log-level-based counterparts will utilize this provider by merging the
-provided metadata with the one-off metadata within the log statement:
+> NOTE: See [SE-0311](https://github.com/apple/swift-evolution/blob/main/proposals/0311-task-locals.md) for an explanation of task-local values in Swift.
+
+Once configured, the `log` function and its log-level-based counterparts will utilize this provider by merging the provided metadata with the one-off metadata within the log statement. 
+
+Since `Baggage` is propagaged as a task-local, we do not change any of the existing method signatures, and the baggage is picked up inside the log method:
+
+```swift
+extension Logger {
+    @inlinable
+    func log(level: Logger.Level,
+             _ message: @autoclosure () -> Logger.Message,
+             metadata: @autoclosure () -> Logger.Metadata? = nil,
+             source: @autoclosure () -> String? = nil,
+             file: String = #fileID, function: String = #function, line: UInt = #line)
+    {
+        if self.logLevel <= level {
+            let baggage: Baggage? = Baggage.current
+            self.handler.log(level: level,
+                             message: message(),
+                             metadata: metadata(),
+                             baggage: Baggage.current, // picks up the task-local 🧳
+                             source: source() ?? Logger.currentModule(fileID: (file)),
+                             file: file, function: function, line: line)
+        }
+    }
+}
+```
+
+Note, that the task-local lookup is only performed when the log message is about to be emitted. No task-local or thread-local lookups are performed when the log level is lower than the level of the current statement.
+
+We also introduce another overload of the log (and trace/debug/info/notice/warning/error/critical) method, which allows libraries to pass a baggage explicitly if they need to (or, disable baggage value extraction by passing `nil` as the value):
 
 ```swift
 extension Logger {
-    // unrelated code location parameters left out
+    @inlinable
     func log(level: Logger.Level,
-        _ message: @autoclosure () -> Logger.Message,
-        metadata: @autoclosure () -> Logger.Metadata? = nil,
-        baggage: Baggage? = .current) {
```

Comment:
Thank you for writing it up, this made me realize we can't have it here :) 
Since we want to avoid the task-local and thread-local lookup whenever we can, we have to put the picking up into the body, since it's dependent on the log level

Reply:
Yup yeah, it's the same API as before